lib/honeybadger/backend/server.rb

```diff
@@ -48,8 +49,103 @@ def check_in(id)
+      #   backend.get_check_in('1234', 'ajdja")
+      #
+      # @param [String] project_id The unique project id
+      # @param [String] id The unique check_in id
+      # @returns [CheckIn] or nil if check_in is not found
+      # @raises CheckInSyncError on error
+
+      def get_check_in(project_id, id)
+        response = Response.new(@http.get("/v2/projects/#{project_id}/check_ins/#{id}", personal_auth_headers))
+        if response.success?
+          return CheckIn.from_remote(project_id, JSON.parse(response.body))
+        else
+          if response.code == 404
+            return nil
+          end
+        end
+        raise CheckInSyncError.new "Fetching CheckIn failed (Code: #{response.code}) #{response.body}"
+      end
+
+      # Get check_ins by project
+      # @example
+      #   backend.get_check_ins('1234')
+      #
+      # @param [String] project_id The unique project id
+      # @returns [Array<CheckIn>] All checkins for this project
+      # @raises CheckInSyncError on error
+      def get_check_ins(project_id)
+        response = Response.new(@http.get("/v2/projects/#{project_id}/check_ins", personal_auth_headers))
+        if response.success?
+          all_check_ins = JSON.parse(response.body)["results"]
+          return all_check_ins.map{|cfg| CheckIn.from_remote(project_id, cfg) }
+        end
+        raise CheckInSyncError.new "Fetching CheckIns failed (Code: #{response.code}) #{response.body}"
+      end
+
+      # Create check_in on project
+      # @example
+      #   backend.create_check_in('1234', check_in)
+      #
+      # @param [String] project_id The unique project id
+      # @param [CheckIn] check_in_config A CheckIn object encapsulating the config
+      # @returns [CheckIn] A CheckIn object additionally containing the id
+      # @raises CheckInSyncError on error
+      def create_check_in(project_id, check_in_config)
+        response = Response.new(@http.post("/v2/projects/#{project_id}/check_ins", check_in_config.to_json, personal_auth_headers))
+        if response.success?
+          return CheckIn.from_remote(project_id, JSON.parse(response.body))
+        end
+        raise CheckInSyncError.new "Creating CheckIn failed (Code: #{response.code}) #{response.body}"
+      end
+
+      # Update check_in on project
+      # @example
+      #   backend.update_check_in('1234', 'eajaj', check_in)
+      #
+      # @param [String] project_id The unique project id
+      # @param [String] id The unique check_in id
+      # @param [CheckIn] check_in_config A CheckIn object encapsulating the config
+      # @returns [CheckIn] updated CheckIn object
+      # @raises CheckInSyncError on error
+      def update_check_in(project_id, id, check_in_config)
+        response = Response.new(@http.put("/v2/projects/#{project_id}/check_ins/#{id}", check_in_config.to_json, personal_auth_headers))
+        if response.success?
+          return CheckIn.from_remote(project_id, JSON.parse(response.body))
```

Comment:
Did you test this route? If I remember well, it does not return a body; only a HTTP 204 (no content).